ibm/data_source_ibm_satellite_host_script.go

```diff
@@ -0,0 +1,131 @@
+				Set:         schema.HashString,
+				Description: "List of labels for the attach host",
+			},
+			"host_provider": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"script_dir": {
+				Description: "The directory where the satellite attach host script to be downloaded. Default is home directory ",
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+			},
+			"host_script": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "Attach host script content",
+			},
+		},
+	}
+}
+
+func dataSourceIBMSatelliteAttachHostScriptRead(d *schema.ResourceData, meta interface{}) error {
+	location := d.Get("location").(string)
+	hostProvider := d.Get("host_provider").(string)
+	scriptDir := d.Get("script_dir").(string)
+
+	satClient, err := meta.(ClientSession).SatelliteClientSession()
+	if err != nil {
+		return err
+	}
+
+	bxClient, err := meta.(ClientSession).BluemixSession()
+	if err != nil {
+		return err
+	}
+
+	getSatLocOptions := &kubernetesserviceapiv1.GetSatelliteLocationOptions{
+		Controller: &location,
+	}
+
+	locData, _, err := satClient.GetSatelliteLocation(getSatLocOptions)
+	if err != nil {
+		return err
+	}
+
+	// script labels
+	labels := make(map[string]string)
+	if _, ok := d.GetOk("labels"); ok {
+		l := d.Get("labels").(*schema.Set)
+		labels = flattenHostLabels(l.List())
+	}
+
+	if len(scriptDir) == 0 {
+		scriptDir, err = homedir.Dir()
+		if err != nil {
+			return fmt.Errorf("Error fetching homedir: %s", err)
+		}
+	}
+	scriptDir, _ = filepath.Abs(scriptDir)
+	scriptPath := filepath.Join(scriptDir, "addHost.sh")
+
+	//Generate script
+	createRegOptions := &kubernetesserviceapiv1.AttachSatelliteHostOptions{}
+	createRegOptions.Controller = locData.ID
+	createRegOptions.Labels = labels
+	authMap := map[string]string{"Authorization": bxClient.Config.IAMAccessToken}
+	createRegOptions.Headers = authMap
+
+	resp, err := satClient.AttachSatelliteHost(createRegOptions)
+	if err != nil {
+		return fmt.Errorf("Error Generating Satellite Registration Script: %s", err)
+	}
+
+	lines := strings.Split(string(resp), "\n")
+	for i, line := range lines {
+		if strings.Contains(line, "API_URL=") {
+			i = i + 1
+			if strings.ToLower(hostProvider) == "aws" {
+				lines[i] = "yum update -y\nyum-config-manager --enable '*'\nyum repolist all\nyum install container-selinux -y"
+			} else if strings.ToLower(hostProvider) == "ibm" {
+				lines[i] = "subscription-manager refresh\nsubscription-manager repos --enable=*\n"
+			} else if strings.ToLower(hostProvider) == "azure" {
+				lines[i] = "# Grow the TMP LV\nlvextend -L+10G /dev/rootvg/tmplv\nxfs_growfs /dev/rootvg/tmplv\nyum update -y\nyum-config-manager --enable '*'\nyum repolist all\nyum install container-selinux -y"
+			}
+		}
+	}
+
+	scriptContent := strings.Join(lines, "\n")
+	err = ioutil.WriteFile(scriptPath, []byte(scriptContent), 0644)
```

Comment:
Does this close file after writing?

Reply:
opening and closing of the file handled by ioutil itself. No need to close explicitly.